crates/full-node/sov-sequencer/src/preferred/replica_sync_task.rs

```diff
@@ -251,18 +240,47 @@ where
                         let parsed_notification = EventsNotificationPayload::parse_csv(payload)
                             .map_err(|e| anyhow!("Failed to parse CSV notification payload: {e}"))?;
 
+                        let replayed_sequence_number = *replay_receiver.borrow();
+                        match replayed_sequence_number {
+                            // Node is not synced yet - ignore all incoming events
+                            None => {
+                                trace!("Replica receiving event but node is not synced yet, ignoring. Event: {payload}");
+                                continue;
+                            }
+                            // We are indeed synced. Is the last replayed batch ahead of our event
+                            // stream?
+                            Some(last_replayed_sequence_number) => {
+                                if last_fully_processed_batch.is_none_or(|seq| seq < last_replayed_sequence_number) {
+                                    // Query the batch_close event with this sequence number to find the event_id
+                                    // Set our latest_received_event_id to that event's id so we continue from there
+                                    let new_latest_event_id = query_event_id_for_batch_close(query_pool, last_replayed_sequence_number).await?;
+                                    if latest_received_event_id.is_some_and(|latest| latest >= new_latest_event_id) {
+                                        // Return an error rather than using assert! - this way the
+                                        // sequencer will call exit_rollup() cleanly
+                                        anyhow::bail!("Sanity check failed - update_state fast-forwarded our sequence number, but this would not have fast-forwarded our event ID");
+                                    }
+                                    latest_received_event_id = Some(new_latest_event_id);
```

Comment:
Is that ok that we override `latest_received_event_id` with `Some(parsed_notification.event_id)` in line 286

Reply:
Yes that should be fine - we use this value to check for backfill for events between `new_latest_event_id` and `parsed_notification.event_id`. Once we've inserted the backfill, the latter is the correct value